src/components/views/room_settings/AliasSettings.js

```diff
@@ -65,11 +70,21 @@ module.exports = React.createClass({
 
         if (canonicalAliasEvent) {
             state.canonicalAlias = canonicalAliasEvent.getContent().alias;
+            state.initialCanonicalAlias = state.canonicalAlias;
         }
 
         return state;
     },
 
+    refreshHasAliases: function() {
+        const localDomain = MatrixClientPeg.get().getDomain();
+        this.props.onRefreshHasAliases(
+            this.state.domainToAliases[localDomain].length > 0 ||
```

Comment:
For the `"before each" hook for "should not set when no setting is changed"` test, `this.state.domainToAliases[localDomain]` ends up being undefined here.